Worker/src/Unified/TaskSubmitterWorkerService.cs

```diff
@@ -66,59 +67,62 @@ public TaskSubmitterWorkerService()
   /// <summary>
   /// </summary>
   /// <param name="loggerFactory">The factory logger to create logger</param>
-  public TaskSubmitterWorkerService(ILoggerFactory loggerFactory)
+  public TaskSubmitterWorkerService(ILoggerFactory? loggerFactory)
   {
     LoggerFactory = loggerFactory;
 
-    Logger = loggerFactory.CreateLogger(GetType()
-                                          .Name);
+    Logger = loggerFactory?.CreateLogger(GetType()
+                                           .Name);
   }
 
   /// <summary>
   ///   Get access to Logger with Logger.LoggingScope.
   /// </summary>
-  public ILogger Logger { get; set; }
+  public ILogger? Logger { get; set; }
 
   /// <summary>
   ///   Get or Set SubSessionId object stored during the call of SubmitTask, SubmitSubTask,
   ///   SubmitSubTaskWithDependencies or WaitForCompletion, WaitForSubTaskCompletion or GetResults
   /// </summary>
-  public Session SessionId { get; set; }
+  public Session SessionId { get; set; } = new()
+                                           {
+                                             Id = "BadSessionId",
+                                           };
 
   /// <summary>
   ///   Property to retrieve the sessionService previously created
   /// </summary>
-  internal SessionPollingService SessionService { get; set; }
+  internal SessionPollingService? SessionService { get; set; }
 
   /// <summary>
   ///   Map between ids of task and their results id after task submission
   /// </summary>
   public Dictionary<string, string> TaskId2OutputId
-    => SessionService.TaskId2OutputId;
+    => SessionService?.TaskId2OutputId ?? throw new NoNullAllowedException(nameof(SessionService));
```

Comment:
If this service was properly initialized in the constructor (base or derived class), this issue would not happen.
Moreover, with this error, you expose an implementation detail : the fact that `SessionService` due to an implementation choice without giving any clue on how to avoid this situation.